code/modules/mob/dead/observer/observer.dm

```diff
@@ -365,6 +365,10 @@ Works together with spawning an observer, noted above.
 
 	ghost.set_huds_from_prefs()
 
+	//if zombie turn your body back to the free mob pool
+	if(iszombie(src))
+		free_for_ghosts()
```

Comment:
i need to add a check so that zombie without their head don't become free mob...
i tried multiple stuff nothing work....